guides/gatherers.md

```diff
@@ -65,4 +66,44 @@ facts:
     argument: nodelist.node
 ```
 
-For extra information refer to [trento-project/agent/../gatherers/corosyncconf_test.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/corosyncconf_test.go)
+For more information refer to [trento-project/agent/../gatherers/corosyncconf_test.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/corosyncconf_test.go)
+
+### hosts (/etc/hosts)
+
+**Name:** `hosts`
+
+This gatherer allows accessing the hostnames that are resolvable through `/etc/hosts`. It
+does **not** use domain resolution in any way but instead directly parses the file.
+
+It allows one argument to be specified or none at all:
+ - When a hostname is provided as an argument, the gatherer will return an array of IPv4 and/or IPv6 addresses.
+ - When no argument is provided, the gatherer will return a map with hostname as keys and arrays with IPv4 and/or IPv6 addresses.
+
+```
+hosts_related_fact = hosts(example.com)
+```
+
+Sample arguments
+| name                                 | Return value          
+| ------------------------------------ | --------------------------------------------------------
+| `localhost`                          | list of IPs resolving `localhost` e.g. `["127.0.0.1", "::1"]`
+| `node1`                              | list of IPs resolving `node1` e.g. `["172.22.0.1"]`
+| `nil` or `""`                        | map with hostnames and IP addresses e.g. `{"localhost": ["127.0.0.1", "::1"], "node1": ["172.22.0.1"]}`
```

Comment:
I am afraid this `nil` or `""` detail is not really relevant from a specification perspective, and it might be misleading because we do not really pass `nil` or `""`, we just don't provide the argument altogether.

We could rephrase to `No argument provided` (or similar), as the following yaml example shows.

wdyt?